src/resolvers/userResolver.js

```diff
@@ -134,6 +134,62 @@ const userResolver = {
       );
     },
 
+    changeUsername: async(_, { userInfo }, { req }) => {
+      const { newUsername } = userInfo;
+      const oldUser = await User.findById(req.session.userId);
+      const oldUsername = oldUser.username;
+
+      if(oldUsername === newUsername) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "old and new username are same",
+            },
+          ],
+        };
+      }
+
+      let user = await User.findOne({ username: oldUsername });
+      if(!user) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "username does not exists",
+            },
+          ],
+        };
+      }
+
+      try {
+        await User.updateOne(
+          { 
+            username: oldUsername 
+          }, 
+          { $set: 
+            { 
+              username: newUsername 
+            } 
+        });
+
+      } catch(error) {
+        if (error.code === 11000 && Object.keys(error.keyValue)[0] === "username") {
+          return {
+            errors: [
+              {
+                field: "username",
+                message: "username already exists",
+              },
+            ],
+          };
+        }
+      }
+
+      let updatedUser = await User.findOne({ username: newUsername });
```

Comment:
Always try to use `findById()` whenever possible because it has a better time complexity in MongoDB. Since here we have the `userId` of the user available therefore we can use `findById()`.